### PR TITLE
Keep the status action buttons at their position regardless of the counter size

### LINKED ISSUE #32030: Reply-All button shifts action bar icons to the side
### Steps to reproduce the problem

Mastodon web interface. One action-bar with a 0-comments thread-icon, and one action bar with a 1 comment, single-thread icon. The remaining icons (share, fav, save, menu) are offset by a couple of pixels.

Same, but on 4.3 mastodon.social (as of creating the ticket)
image

### Expected behaviour

The action-bar is a static grid with space allocated for each button. Changing the width of one element, shouldn't affect the other elements (within reasonable limits).

### Actual behaviour

As the reply-all button is wider, than the reply button, the action bar icons should not realign.

### Detailed description

_No response_

### Mastodon instance

mastodon.social

### Mastodon version

v4.2.12 & v4.3.0-nightly.2024-09-23

### Browser name and version

Edge 129

### Operating system

Windows 11, 23h2

### Technical details

_No response_

### PR DESCRIPTION
**Pull request by renchap**
image

image

I also added a small left padding to the button when there is a counter, it stopped right at the text and this was not consistent with the space around the icon.

Fixes #32030

### CODE REVIEW

File: app/javascript/styles/mastodon/components.scss [hunk 1]
```diff
@@ -299,6 +299,10 @@
     }
   }
 
+  &--with-counter {
+    padding-right: 4px;
```
Comment (ClearlyClaire) [inline app/javascript/styles/mastodon/components.scss:303]: This should be changed to account for RTL layouts:
```suggestion
    padding-inline-end: 4px;
```

**Review by ClearlyClaire — COMMENTED** [review]
Is the extra layer of wrapping necessary?

**Comment by renchap** [comment]
> Is the extra layer of wrapping necessary?

Yes, otherwise the buttons will take the largest width possible, which makes the whole area clickable

**Review by ClearlyClaire — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
> > Is the extra layer of wrapping necessary?
> 
> Yes, otherwise the buttons will take the largest width possible, which makes the whole area clickable

Ah, yes. This is a bit annoying, but I guess this is fine.

**Pull request merged**